lib/common/common/src/cpu.rs

```diff
@@ -0,0 +1,47 @@
+#[cfg(target_os = "linux")]
+use thiserror::Error;
+#[cfg(target_os = "linux")]
+use thread_priority::{set_current_thread_priority, ThreadPriority, ThreadPriorityValue};
+
+#[derive(Error, Debug)]
+#[cfg(target_os = "linux")]
+pub enum ThreadPriorityError {
+    #[error("Failed to set thread priority: {0:?}")]
+    SetThreadPriority(thread_priority::Error),
+    #[error("Failed to parse thread priority value: {0}")]
+    ParseNice(&'static str),
+}
+
+/// On Linux, make current thread lower priority (nice: 10).
+#[cfg(target_os = "linux")]
+pub fn linux_low_thread_priority() -> Result<(), ThreadPriorityError> {
+    // 25% corresponds to a nice value of 10
+    set_linux_thread_priority(25)
+}
+
+/// On Linux, make current thread high priority (nice: -10).
+///
+/// # Warning
+///
+/// This is very likely to fail because decreasing the nice value requires special privileges. It
```

Comment:
Did you check if it works with Docker?

Reply:
I did not, but I expect the same behavior. I'll give that a spin now.

Reply:
Yes, the behavior is the same. So that means things work fine :+1: